### PR TITLE
#801 Trying to load mapper implementation types via the mapper interf…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…ace's loader first

### CODE REVIEW

File: core-common/src/main/java/org/mapstruct/factory/Mappers.java [hunk 1]
```diff
@@ -69,32 +71,57 @@ private Mappers() {
             // - clazz is annotated with @Mapper
             //
             // Use privileged action
```
Comment (agudian) [inline core-common/src/main/java/org/mapstruct/factory/Mappers.java]: should we maybe remove these comments here?

Comment (gunnarmorling) [inline core-common/src/main/java/org/mapstruct/factory/Mappers.java]: +1. Opened a new issue for these.

**Comment by agudian** [comment]
Aside from the checkstyle issue, it looks good, also the order of the classloaders makes sense to me :+1: 
Go ahead and push it to master when you're done...

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks for reviewing, @agudian!

**Pull request closed without merging**